ReactiveXComponent/RabbitMq/RabbitMqConnection.cs

```diff
@@ -1,20 +1,102 @@
-﻿using ReactiveXComponent.Configuration;
+﻿using System;
+using System.Threading;
+using RabbitMQ.Client;
+using ReactiveXComponent.Configuration;
 using ReactiveXComponent.Connection;
 
 namespace ReactiveXComponent.RabbitMq
 {
     public class RabbitMqConnection : IXCConnection
     {
-        private readonly XCConfiguration _xcConfiguration;
+        private const string CannotReachBroker = "Cannot connect to broker ";
 
-        public RabbitMqConnection(XCConfiguration configuration)
+        private const int TimeoutConnection = 10000;
+
+        private bool _disposed;
+        private IConnection _connection;
+        private ConnectionFactory _factory;
+
+        private readonly IXCConfiguration _xcConfiguration;
+
+        public RabbitMqConnection(IXCConfiguration configuration)
         {
             _xcConfiguration = configuration;
+            Init(configuration.GetBusDetails());
+            _connection = CreateConnection();
+        }
+
+        private void Init(BusDetails busDetails)
+        {
+            try
+            {
+                _factory = new ConnectionFactory()
+                {
+                    UserName = busDetails.Username ?? "guest",
+                    Password = busDetails.Password ?? "guest",
+                    VirtualHost = ConnectionFactory.DefaultVHost,
+                    HostName = busDetails.Host,
+                    Port = busDetails.Port,
+                    Protocol = Protocols.DefaultProtocol
+                };
+            }
+            catch (Exception e)
+            {
+                throw new Exception("RabbitMQ Connection init failed", e);
+            }
+        }
+
+        private IConnection CreateConnection()
+        {
+            if (_factory != null)
+            {
+                var lockEvent = new AutoResetEvent(false);
+                string errorMessage = null;
+                ThreadPool.QueueUserWorkItem((obj) =>
```

Comment:
I don't understand your implementation with the ThreadPool
Perhaps you want to also provide a CreateConnectionAsync